File: src/go/08_regexp/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	infile := "access_log"
	r, err := os.Open(infile)
	if err != nil {
		panic(err)
	}
	defer r.Close()

	sc := bufio.NewScanner(r)
	for sc.Scan() {
		line := sc.Text()

		// TODO

		ip := ""
		fmt.Println(ip)
	}
	if err := sc.Err(); err != nil {
		panic(err)
	}
}

File: src/clojure/08_regexp/main.clj
(require '[clojure.java.io :as io]
         '[clojure.string :as str])

(with-open [fin (io/reader "access_log")]
  (doseq [line (line-seq fin)]
    (-> line
        (str/replace #"^([^\s]+) .*" "$1")
        println)))
